### ISSUE TITLE
[java] UseUnderscoresInNumericLiterals in annotations

### ISSUE THREAD

**Issue by mattnelson** [pmd/pmd#1528]
**Affects PMD Version:** 

**Rule:**

UseUnderscoresInNumericLiterals

**Description:**

Detect numeric literals in annotations.

**Code Sample demonstrating the issue:**

```java
@NumericAnnotation(1_000_000L)
class AnnotatedClass {
}
```

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
@mattnelson welcomed.

I fear your report is incomplete, as it's impossible for us to understand what the problem is.

The rule `UseUnderscoresInNumericLiterals` does check for **all literals**, no matter where they are.

```java
@NumericAnnotation(1000000L) // gets flagged, doesn't use underscores in spite of exceeding 4 digits
@NumericAnnotation(1_000_000L) // this is ok
class AnnotatedClass {
}
```

I'm closing this for the time being. Feel free to reopen after editing it to clarify the problem you are experiencing.

**Comment by mattnelson** [comment]
> gets flagged, doesn't use underscores in spite of exceeding 4 digits

Thanks, this was the root of my issue. I expected it to fail on 1000, but for the example on the issue I used 1,000,000.

**Comment by jsotuyod** [comment]
@mattnelson awesome!

That minimum ñength is configurable through a property, you can check the docs for more detail: https://pmd.github.io/pmd-6.10.0/pmd_rules_java_codestyle.html#useunderscoresinnumericliterals
